src/extra/gpkg/read.jl

```diff
@@ -0,0 +1,439 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+# GeoPackage Binary Header structure
+struct GeoPackageBinaryHeader
+  magic::UInt16
+  version::UInt8
+  flags::UInt8
+  srs_id::Int32
+  min_x::Float64
+  max_x::Float64
+  min_y::Float64
+  max_y::Float64
+  min_z::Union{Float64,Nothing}
+  max_z::Union{Float64,Nothing}
+  min_m::Union{Float64,Nothing}
+  max_m::Union{Float64,Nothing}
+end
+
+function parse_gpb_header(blob::Vector{UInt8})
+  io = IOBuffer(blob)
+  
+  magic = read(io, UInt16)
+  magic == 0x5047 || error("Invalid GeoPackage binary header")
+  
+  version = read(io, UInt8)
+  flags = read(io, UInt8)
+  
+  srs_id = ltoh(read(io, Int32))
+  
+  envelope_type = flags >> 1 & 0x07
+  has_envelope = envelope_type > 0
+  
+  if has_envelope && envelope_type >= 1
+    min_x = ltoh(read(io, Float64))
+    max_x = ltoh(read(io, Float64))
+    min_y = ltoh(read(io, Float64))
+    max_y = ltoh(read(io, Float64))
+  else
+    # Default envelope when not present
+    min_x = max_x = min_y = max_y = 0.0
+  end
+  
+  # Initialize z and m coordinates based on envelope type
+  if has_envelope && envelope_type >= 2
+    min_z = ltoh(read(io, Float64))
+    max_z = ltoh(read(io, Float64))
+  else
+    min_z = max_z = nothing
+  end
+  
+  if has_envelope && envelope_type >= 3
+    min_m = ltoh(read(io, Float64))
+    max_m = ltoh(read(io, Float64))
+  else
+    min_m = max_m = nothing
+  end
+  
+  header = GeoPackageBinaryHeader(
+    magic, version, flags, srs_id,
+    min_x, max_x, min_y, max_y,
+    min_z, max_z, min_m, max_m
+  )
+  
+  return header, position(io)
+end
+
+function read_coordinate_values(io::IOBuffer, byte_order::UInt8, has_z::Bool, has_m::Bool)
+  read_value = byte_order == 0x01 ? ltoh : ntoh
+  
+  x = read_value(read(io, Float64))
+  y = read_value(read(io, Float64))
+  
+  coords_list = [x, y]
+  
+  if has_z
+    z = read_value(read(io, Float64))
+    push!(coords_list, z)
+  end
+  
+  # Skip M coordinate if present (we don't use it)
+  if has_m
+    read_value(read(io, Float64))
+  end
+  
+  return coords_list
+end
+
+function create_point_from_coords(coords_list::Vector{Float64}, crs_type)
+  x, y = coords_list[1], coords_list[2]
+  
+  if isnan(x) && isnan(y)
+    return nothing
+  end
+  
+  # Handle different CRS types properly
+  if isnothing(crs_type)
+    # No CRS specified - use raw coordinates as Cartesian
+    return Point(coords_list...)
+  end
```

Comment:
Please remove this branch. I believe the gpkg file always has a valid CRS type. If it doesn't have a valid CRS, then the `crs_type` must be set to a default outside this function. The function `create_point_from_coords` should assume that the argument `crs_type` is always a `CRS` subtype.